# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1552337370 0
#      Mon Mar 11 20:49:30 2019 +0000
# Node ID e247ecdee41da2c722cae213b78ce215044a0025
# Parent  e201cf0c736028bcaf565e71dee213ea53c50051
allow building without QGuiApplication to work again

* shortcut-manager.cc (enter_shortcut::enter_shortcut):
Include QApplication instead of QGuiApplication.
Use QApplication::keyboardModifiers if HAVE_QGUIAPPLICATION is not
defined.

diff --git a/libgui/src/shortcut-manager.cc b/libgui/src/shortcut-manager.cc
--- a/libgui/src/shortcut-manager.cc
+++ b/libgui/src/shortcut-manager.cc
@@ -23,17 +23,17 @@ along with Octave; see the file COPYING.
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <QtCore>
 #include <QMessageBox>
 #include <QDebug>
 #include <QGridLayout>
-#include <QGuiApplication>
+#include <QApplication>
 #include <QVBoxLayout>
 #include <QDialogButtonBox>
 #include <QKeySequence>
 #include <QPushButton>
 #include <QLineEdit>
 #include <QCheckBox>
 #include <QHeaderView>
 #include <QAction>
@@ -65,17 +65,21 @@ namespace octave
 
     if (e->type () == QEvent::KeyPress)
       {
         int key = e->key ();
 
         if (key == Qt::Key_unknown || key == 0)
           return;
 
+#if defined (HAVE_QGUIAPPLICATION)
         Qt::KeyboardModifiers modifiers = QGuiApplication::keyboardModifiers (); //e->modifiers ();
+#else
+        Qt::KeyboardModifiers modifiers = QApplication::keyboardModifiers (); //e->modifiers ();
+#endif
 
         if (m_shift_modifier || (modifiers & Qt::ShiftModifier))
           key += Qt::SHIFT;
         if (modifiers & Qt::ControlModifier)
           key += Qt::CTRL;
         if (modifiers & Qt::AltModifier)
           key += Qt::ALT;
         if (modifiers & Qt::MetaModifier)
